fix: Odata generation doesn't fail if basePath is not set (#4663)
SAP/cloud-sdk-js: .changeset/shy-days-dance.md

---
"@sap-cloud-sdk/generator": minor
---

> [Improvement] If basePath cannot be determined, instead of generation failing altogether, basePath is set to `/`, and a warning message is logged.

Reviewer: ```suggestion
[Improvement] If `basePath` cannot be determined, it is set to `/` and generation no longer fails.
```
